PyRDF/Node.py

```diff
@@ -5,44 +5,140 @@
 
 class Node(object):
     """
-    A Class that represents
-    a graph Node
+    A Class that represents a node in RDataFrame
+    operations graph. A Node houses an operation
+    and has references to child nodes. For details
+    on the types of operations supported, try :
+
+    import PyRDF
+    help(PyRDF.Operation)
+
+    Attributes
+    ----------
+    operation
+        The operation that this Node represents. This
+        could be `None`.
+
+    next_nodes
+        A list of `Node` objects which represent the
+        children nodes connected to the current node.
+
+    value
+        The computed value after executing the operation in
+        the current node for a particular PyRDF graph. This
+        is permanently `None` for transformation nodes and
+        the action nodes get a `RResultPtr` after event-loop
+        execution.
 
     """
     def __init__(self, _get_head, operation):
-        
+        """
+        Creates a new `Node` based on the 'operation'.
+
+        Parameters
+        ----------
+        _get_head : function
```

Comment:
Perhaps this parameter should not have the leading `_`, since it has to appear (as parameter) in the documentation.

Reply:
And this as well. Can I include this as a part of the clean up PR ?

Reply:
Yes sure